src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,649 @@
+            sb.append(" costCenterChartCode ").append(KRADConstants.SINGLE_QUOTE).append(costCenterChartCode).append(KRADConstants.SINGLE_QUOTE);
+            sb.append(" costCenterAccountNumber ").append(KRADConstants.SINGLE_QUOTE).append(costCenterAccountNumber).append(KRADConstants.SINGLE_QUOTE);
+            sb.append(" costCenterSubAccountNumber ").append(KRADConstants.SINGLE_QUOTE).append(costCenterSubAccountNumber).append(KRADConstants.SINGLE_QUOTE);
+            sb.append(" costCenterObjectCode ").append(KRADConstants.SINGLE_QUOTE).append(costCenterObjectCode).append(KRADConstants.SINGLE_QUOTE);
+            sb.append(" costCenterSubObjectCode ").append(KRADConstants.SINGLE_QUOTE).append(costCenterSubObjectCode).append(KRADConstants.SINGLE_QUOTE);
+            sb.append(" costCenterProjectCode ").append(KRADConstants.SINGLE_QUOTE).append(costCenterProjectCode).append(KRADConstants.SINGLE_QUOTE);
+            sb.append(" costCenterObjectRefId ").append(KRADConstants.SINGLE_QUOTE).append(costCenterObjectRefId).append(KRADConstants.SINGLE_QUOTE);
+            LOG.debug(sb.toString());
+        }
+    }
+
+    private void updateTransactionDTOAccountNumber(AmazonAccountDetail amazonAccountDetail, TransactionDTO transactionDTO) {
+        Account account = determineWhichAccountToUse(amazonAccountDetail.getCostCenter(), amazonAccountDetail.getKfsAccount());
+        if (account != null) {
+            updateTransactionDTOChartAccountObjectCodes(transactionDTO, account);
+        } else {
+            transactionDTO.setFailingError(true);
+        }
+    }
+    
+    private void updateTransactionDTOAccountNumbers(AmazonAccountDetail amazonAccountDetail, TransactionDTO transactionDTO, String costCenterAccountNumber,
+            String costCenterSubAccountNumber) {
+        Account account = determineWhichAccountToUse(costCenterAccountNumber, amazonAccountDetail.getKfsAccount());
+        if (account != null) {
+            updateTransactionDTOChartAccountObjectCodes(transactionDTO, account);
+            if (StringUtils.isNotBlank(costCenterSubAccountNumber)) {
+                SubAccount subAccount = buildSubAccount(account, costCenterSubAccountNumber);
+                if (subAccount != null) {
+                    transactionDTO.setSubAccountNumber(subAccount.getSubAccountNumber());
+                } else {
+                    transactionDTO.setFailingError(true);
+                }
+            }
+        } else {
+            transactionDTO.setFailingError(true);
+        }
+    }
+    
+    private Account determineWhichAccountToUse(String costCenter, String kfsAccountString) {
+        Account costAccount = buildAccount(costCenter);
+        if (validAccount(costAccount)) {
+            return costAccount;
+        }
+        Account kfsAccount = buildAccount(kfsAccountString);
+        if (validAccount(kfsAccount)) {
+            return kfsAccount;
+        }
+        return null;
```

Comment:
I wonder if you could avoid returning null by just returning a new Account, or something like that.